### ISSUE TITLE
field_instances

### ISSUE THREAD

**Issue by gausie** [UCLU/appointment#3]
@pjcdawkins could you possibly help me out?

I'm trying to create other slot types and so need to add the fields into `modules/slot_fields/appointment_slot_fields.features.field_instance.inc`. Looking at it it's clearly something that has been exported from Features, but I can't work out where it was actually created.

How do I add fields to custom slot types?

Thanks!

**Comment by pjcdawkins** [comment]
Can you create the fields in the UI, and export the field instance(s) into your own feature module?

`drush fe uclu_appointments field_instance:appointment_slot-field_my_custom_field`

**Comment by gausie** [comment]
I can't create fields in the UI at all, that's why I'm confused.

**Comment by gausie** [comment]
[This](https://github.com/UCLU/uclu.org/tree/feature/appointment_fixes) is where I'm working on it on the u.o end, but I've just been manually fiddling.

**Comment by pjcdawkins** [comment]
Hm. Maybe it doesn't expose a good fields admin UI... do you get something at this path? `admin/structure/appointments/slots/types/<type>`

and probably: `admin/structure/appointments/slots/types/<type>/fields`

**Comment by gausie** [comment]
I just get "You do not have any administrative items."

**Comment by gausie** [comment]
Also I've just noticed [uclu_visa](https://github.com/UCLU/uclu.org/tree/develop/sites/all/modules/uclu/uclu_visa), was this finished?

**Comment by gausie** [comment]
/fields works though. Why would that be?

**Comment by gausie** [comment]
Ah I see what's going on. There's nothing to land on but it can still edit the Fields. However, we get two sets of each menu item, one for each of the types. ![image](https://cloud.githubusercontent.com/assets/179758/6371185/e3dc8f6c-bcf0-11e4-85a2-881721c7266b.png)

**Comment by pjcdawkins** [comment]
Yes, these menu items are set up automatically by the Drupal field API... they are pretty annoying and hard to sort out in a lightweight way. At the time, I considered it good enough for me to use as a developer... and didn't document it.

The cause is here:
https://github.com/UCLU/appointment/blob/7.x-1.x/modules/slot/appointment_slot.module#L49

and I assume it could be sorted out with an "Entity API" module admin UI controller class, cf. election_candidate:
http://cgit.drupalcode.org/election/tree/election_candidate/election_candidate.module#n70

**Comment by gausie** [comment]
Will look at this all thanks!!
